### PR TITLE
feat: DBC and HeeUser requests from TSS

### PR DESCRIPTION
**Pull request by ReubenRobertsHEE**
TIS21-6272

### CODE REVIEW

**Review by JosephKelly — CHANGES_REQUESTED** [review]
One import to change.
I'm happy enough with the rest if you want to leave them... if you're up for addressing the code smell too, that would be appreciated!

File: src/test/java/uk/nhs/tis/sync/mapper/HeeUserMapperTest.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.transformuk.hee.tis.profile.dto.RoleDTO;
+import com.transformuk.hee.tis.profile.service.dto.HeeUserDTO;
+import com.transformuk.hee.tis.profile.service.dto.UserProgrammeDTO;
+import com.transformuk.hee.tis.profile.service.dto.UserTrustDTO;
+import java.util.HashSet;
+import java.util.Set;
+import org.junit.Test;
```
[the remark this answers is not in the document]
Comment (ReubenRobertsHEE) [reply src/test/java/uk/nhs/tis/sync/mapper/HeeUserMapperTest.java:10]: Is this a more general upgrade for the project? I can't see any other tests that use JUnit5.

[the remark this answers is not in the document]
Comment (JosephKelly) [reply src/test/java/uk/nhs/tis/sync/mapper/HeeUserMapperTest.java:10]: This test uses JUnit5 Jupiter annotations :smile: , i.e. `@ParameterizedTest` was introduced with JUnit5.

[the remark this answers is not in the document]
Comment (ReubenRobertsHEE) [reply src/test/java/uk/nhs/tis/sync/mapper/HeeUserMapperTest.java]: Sorry, I was being obtuse. Fixed now

File: src/main/java/uk/nhs/tis/sync/dto/DmsDtoType.java [hunk 2]
```diff
@@ -86,4 +87,13 @@ public static DmsDtoType fromDto(Object dto) {
     }
     return null;
   }
+
+  /**
+   * Define the schema names.
+   */
+  private static class Schemas {
```
Comment (JosephKelly) [inline src/main/java/uk/nhs/tis/sync/dto/DmsDtoType.java:94]: Noice!

**Review by JosephKelly — DISMISSED** [review]

**Review by JosephKelly — APPROVED** [review]
:joy:

File: src/main/java/uk/nhs/tis/sync/dto/DmsDtoType.java (hunk 2 above)
Comment (ReubenRobertsHEE) [reply src/main/java/uk/nhs/tis/sync/dto/DmsDtoType.java:94]: Sorry, one last code smell straggler came through after your approval 🐑

**Comment by ReubenRobertsHEE** [comment]
Testing on staging shows there is a permission issue for the HeeUser requests from the Profile service. I'm looking into this now.

e.g.:

```
2024-07-15 08:11:40.048 ERROR 1 --- [   Thread-12274] u.n.tis.sync.service.DataRequestService  : 403 Forbidden: [Access is denied]
org.springframework.web.client.HttpClientErrorException$Forbidden: 403 Forbidden: [Access is denied]
	at org.springframework.web.client.HttpClientErrorException.create(HttpClientErrorException.java:109) ~[spring-web-5.2.7.RELEASE.jar:5.2.7.RELEASE]
	at org.springframework.web.client.DefaultResponseErrorHandler.handleError(DefaultResponseErrorHandler.java:184) ~[spring-web-5.2.7.RELEASE.jar:5.2.7.RELEASE]
	at org.springframework.web.client.DefaultResponseErrorHandler.handleError(DefaultResponseErrorHandler.java:125) ~[spring-web-5.2.7.RELEASE.jar:5.2.7.RELEASE]
	at org.springframework.web.client.ResponseErrorHandler.handleError(ResponseErrorHandler.java:63) ~[spring-web-5.2.7.RELEASE.jar:5.2.7.RELEASE]
	at org.springframework.web.client.RestTemplate.handleResponse(RestTemplate.java:782) ~[spring-web-5.2.7.RELEASE.jar:5.2.7.RELEASE]
	at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:740) ~[spring-web-5.2.7.RELEASE.jar:5.2.7.RELEASE]
	at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:674) ~[spring-web-5.2.7.RELEASE.jar:5.2.7.RELEASE]
	at org.springframework.web.client.RestTemplate.exchange(RestTemplate.java:612) ~[spring-web-5.2.7.RELEASE.jar:5.2.7.RELEASE]
	at com.transformuk.hee.tis.profile.client.service.impl.ProfileServiceImpl.getSingleAdminUser(ProfileServiceImpl.java:188) ~[profile-client-2.8.1.jar:na]
```

**Pull request merged**